src/sentry/grouping/ingest/grouphash_metadata.py

```diff
@@ -1,24 +1,106 @@
 from __future__ import annotations
 
+import logging
+
+from sentry.eventstore.models import Event
+from sentry.grouping.variants import BaseVariant
 from sentry.models.grouphash import GroupHash
-from sentry.models.grouphashmetadata import GroupHashMetadata
+from sentry.models.grouphashmetadata import GroupHashMetadata, HashBasis
+from sentry.models.project import Project
+
+logger = logging.getLogger(__name__)
+
+GROUPING_METHODS_BY_DESCRIPTION = {
```

Comment:
is there a place where all of these keys are defined?

Reply:
No, because `description` is a property, and it behaves differently depending on the grouping method.

Sometimes it returns a hard-coded string, as the `CustomFingerprintVariant` does here:
https://github.com/getsentry/sentry/blob/81a16b22dea0f22c2f0c33c1887afedaf97596c6/src/sentry/grouping/variants.py#L166-L168

Other times, though, it returns a computed value, as with component variants (which are what's used when grouping by stacktrace, by message, by template, and by security report data), as can be seen here (the first is `ComponentVariant.description` and the second is `GroupingComponent.description`):
https://github.com/getsentry/sentry/blob/81a16b22dea0f22c2f0c33c1887afedaf97596c6/src/sentry/grouping/variants.py#L117-L119
https://github.com/getsentry/sentry/blob/81a16b22dea0f22c2f0c33c1887afedaf97596c6/src/sentry/grouping/component.py#L64-L83

Then there are spots where we modify the description, as we do in `SaltedComponentVariant` (which we use when we have a hybrid fingerprint) :
https://github.com/getsentry/sentry/blob/81a16b22dea0f22c2f0c33c1887afedaf97596c6/src/sentry/grouping/variants.py#L197-L199

So, in short... it's complicated.

Reply:
got it. thank you for the explanation!